### DISCUSSION TITLE
Assembly unloading

### DISCUSSION THREAD

**Discussion by davidsanda** [nxrighthere/UnrealCLR#206] [Help] [answered]
I am using your code as a foundation to run some managed code of my own. The way i structure mine is one assembly, which has multiple dependencies. The main assembly also references framework.
When i run everything, it loads and unloads assemblies as per your runtime code, that apparently loads each assembly and during ManagedCommand load assemblies option and then it unloads it, with exception of framework, where it returns null(default);

This causes the following: 
![image](https://user-images.githubusercontent.com/37351867/142308159-f87e577e-ada0-4463-9934-1aba9f2cc210.png)
When i disable the unloading, everything works just fine. 
Could you please explain the logic of LoadAssemblies command or the intended use?

**Comment by nxrighthere** [comment, marked as answer]
>  then it unloads it, with exception of framework, where it returns null(default);

You have to investigate the call stack and memory using a debugger since, I guess, you modified the code of the plugin's runtime.

> Could you please explain the logic of LoadAssemblies command or the intended use?

It's inspecting assemblies to find the one that references the plugin's framework and then loading it. This logic essentially binds the engine's API to the managed glue.
